sprokit/processes/adapters/embedded_pipeline.cxx

```diff
@@ -116,18 +130,59 @@ class embedded_pipeline::priv
   kwiver::vital::config_block_sptr m_scheduler_config;
   sprokit::scheduler_t m_scheduler;
 
+  embedded_pipeline_extension_sptr m_hooks;
+  std::shared_ptr< embedded_pipeline_extension::context> m_context;
+
 }; // end class embedded_pipeline::priv
 
 
+// ============================================================================
+/*
+ * This implementation of the plugin context provides real access to
+ * our internal data area.
+ *
+ * Other accessors can be added as needed.
+ */
+class real_context
```

Comment:
What's the purpose of having superclass `context` and subclass `real_context`? Do we expect there will ever be a time that we extend `context` with another class?

Reply:
Or is it for the purpose of hiding implementation details?

Reply:
The purpose is to hide the implementation from the public interface.